tokio-postgres/src/client.rs

```diff
@@ -451,6 +450,18 @@ impl Client {
         Ok(Transaction::new(self))
     }
 
+    /// Not part of the public API of this crate.
+    #[doc(hidden)]
+    pub fn cancel_token(&self) -> CancelToken {
```

Comment:
I've tried to avoid having private APIs in tokio-postgres just for postgres's sake. I think it's reasonable for this to just be a public API though. I wonder if we should just deprecate cancel_query and cancel_query_raw?

Reply:
That certainly works for me, if it works for you! The only potential downside is that the current (async-only) API can guarantee at compile time that the connection you're sending the cancellation request on is still alive. The `CancelToken`, of course, can outlive the connection it was created from. Given how racy cancellation is in general, I don't think this is a big deal, but wanted to raise it nonetheless.

Reply:
It cannot guarantee that. `cancel_query` is defined on `Client`, not `Connection`.

Reply:
Oh, good point, I forgot that the client lifetime wasn't tied to the connection lifetime. 👍